core/modules/node/node.types.inc

```diff
@@ -301,6 +301,12 @@ function node_type_form($form, &$form_state, $type = NULL) {
       ),
     );
   }
+  $form['display']['pageless'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Hide full page display'),
+    '#default_value' => $type->settings['pageless'],
+    '#description' => t('If enabled, the URL for this content will return a "Page not found", preventing it from being accessed directly. It may still be placed as a block or within a listing. Users who have the "View page-less content" will still be able to visit the full page URL.'),
```

Comment:
There are two parts of the description, and I think they should be reversed to impart more positivity to use the feature. Right now, I read the first sentence and I get scared maybe not even reading the second part. How about...

"If enabled, this content can be placed as a block or within a listing; however, the URL for this content will return a "Page not found", preventing it from being accessed directly. Users who have the "View page-less content" will still be able to visit the full page URL."

Reply:
Even more positive:

"If enabled, this content can be placed as a block or within a listing but will not be accessible at its URL without proper permissions. Users who have the "View page-less content" permission will still be able to visit the full page URL, while those without this permission will receive a "Page not found" error."

I suspect this has been mentioned before, but "page-less" is a misnomer isn't it? They have pages. Sadly I cant think of anything better though. @jenlampton is good with the words.